pkg/flare/archive_dca.go

```diff
@@ -146,7 +146,7 @@ func getHPAStatus(fb flaretypes.FlareBuilder) error {
 	return fb.AddFile("custommetricsprovider.log", []byte(str))
 }
 
-func getClusterAgentConfigCheck(fb flaretypes.FlareBuilder) error {
+func getClusterAgentConfigCheck(fb flaretypes.FlareBuilder, senderManager sender.DiagnoseSenderManager) error {
```

Comment:
Why are we taking the sender as parameter now ?

Reply:
This change was done by [Remove InitAndStartAgentDemultiplexer from pkg/diagnose/check.go](https://github.com/DataDog/datadog-agent/commit/d907155e0189d45e39d6f38fcc834a6b419259c7). It is a consequence of injecting `diagnosesendermanager.Component` in [diagnoseChecksInCLIProcess](https://github.com/DataDog/datadog-agent/commit/d907155e0189d45e39d6f38fcc834a6b419259c7#diff-c792372d6a3476ee5cf3099091bfd5ce07d3f7223bd4d4627a75f88444a380b8R82)

Reply:
Code updated.